### PR TITLE
Discuss limitations of this technique

### LINKED ISSUE #38: Mention potential downsides of this approach on connectivity
Include experimental results, if we have them

### PR DESCRIPTION
**Pull request by juberti**
Fixes #38

### CODE REVIEW

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 1]
```diff
@@ -174,6 +174,33 @@ An ICE agent that supports mDNS candidates MUST support the situation where the
...
+resolved if the remote endpoint is too many segments away.
+
+When mDNS fails, ICE will attempt to fall back to either NAT hairpin,
+if supported, or TURN relay, if not. As noted in {{IPHandling}}, this may
+result in increased latency and reduced connectivity/increased cost (depending
+on whether the application chooses to use TURN).
+
+Note that backward compatibility does not present a significant issue in this
+situation. When an endpoint that supports mDNS communicates with an endpoint
+that does not, the legacy endpoint will still provide its local IP addresses,
```
Comment (jeroendeb) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: A legacy WebRTC client might attempt a DNS query for the mDNS candidate. End-result is as described, but is an undesired side-effect, potentially delaying the transition to ICE_STATE_COMPLETED.

Not sure if this is worth calling out here.

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Yes, good point. I assume mDNS failure (no record found) happens essentially instantaneously?

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: PTAL

Comment (jeroendeb) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Actually, thinking about it a little further, there may not be that much difference between mDNS and DNS in this case, assume the DNS server considers itself authoritative for the .local. zone.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 2]
```diff
@@ -174,6 +174,36 @@ An ICE agent that supports mDNS candidates MUST support the situation where the
 In this case, the ICE agent MUST take exactly one of the resolved IP addresses and ignore the others.
 The ICE agent SHOULD, if available, use the first IPv6 address resolved, otherwise the first IPv4 address.
 
+Limitations
+===========
+
+With typical ICE, endpoints on the same network will typically be able to
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: How about:
Endpoints on the same network will usually be able to

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 3]
```diff
@@ -174,6 +174,36 @@ An ICE agent that supports mDNS candidates MUST support the situation where the
...
+(depending on whether the application chooses to use TURN).
+
+Note that backward compatibility does not present a significant issue in this
+situation. When an endpoint that supports mDNS communicates with an endpoint
+that does not, the legacy endpoint will still provide its local IP addresses,
+and accordingly a direct connection can still be attempted, even though
+the legacy endpoint cannot resolve the mDNS names provided by the new endpoint.
+In the event the legacy endpoint attempts to resolve mDNS names using Unicast
+DNS, this may cause ICE to take somewhat longer to complete, but should not
+have any effect on connectivity or media latency.
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: Text sounds good to add.
As discussed elsewhere, I would add connection setup time increase in that section.

**Comment by juberti** [comment]
Added new section on latency, as requested by @youennf. PTAL - will plan to merge at EOD if there are no outstanding comments.

**Pull request merged**